cmake/FutilityBuildQuickRefTemplate.rst

```diff
@@ -0,0 +1,48 @@
+==================================================================
+Futility Configure, Build, Test, and Install Quick Reference Guide
+==================================================================
+
+:Author: Benjamin S Collins
+:Contact: bscollin@umich.edu
+
+:Abstract: This document contains quick reference information on how to configure, build, test, and install Futility using the TriBITS CMake build system. The primary audience are developers of Futility that need to configure and build the software.
+
+.. NOTE: Above we can't put in newlines as the RST parser does not allow this.
+
+.. sectnum::
+
+.. contents::
+
+Introduction
+============
+
+Futility description
+
+The Futility Installation Guide provides information on how to clone the
+various Futility repos and getting set up to do a basic build and test of selected
+Futility components.
+
+The set of Futility extra repos is officially defined in the file:
+
+  `Futility/cmake/ExtraRepositoriesList.cmake <cmake/ExtraRepositoriesList.cmake>`_
+
+Configurations of Futility tend to keep complex configuration inside of CMake
+options fragment files and are read in using the TriBITS variable
+``Futility_CONFIGURE_OPTIONS_FILE`` (see `Basic configuration`_ for details).
+Trying to set up a basic configuration for Futility from scratch is very
+difficult.  Thankfully, there are standard configuration for Futility located
+in the directory:
+
+  `Futility/build_scripts`
+
+Futility-specific options
+=========================
+
+Below, configure options specific to Futility are given.  The later sections
+give more generic options that are the same for all TriBITS projects.
+
+First note that the location of ``TRIBITS_BASE_DIR`` in Futility is::
+
+  TRIBITS_BASE_DIR=Futility/cmake/tribits
+
+ToDo: Fill in Futility-specific options.
```

Comment:
Would like to see this be a little more polished before merging. If you're going to create some documentation we should try to have it be complete.

Reply:
This is exactly what has been in MPACT for several years.  I only put it in because you asked me to put in something similar to what was in mpact.

Reply:
honestly.. I don't get the point of this BuildQuickRefTemplate file.  I can't find this in any of the TriBITS or CASL projects other than MPACT.  I'm inclined to say we should just get rid of this file in both Futility and MPACT.  We probably do need to write some build instructions but I feel like its outside of this merge and the underlying ticket.

Reply:
I mean I think the purpose of the quick build ref is just like "hey here are these 5 commands to get you running". It isn't necessary that this needs to be an rst, html, and pdf. It could just be like a README or the first section in the BuildReference.

Either way, I agree addressing the larger issue of updating the installation/build reference documentation deserves its own task. Do you mind making a PHI Kanban ticket for this?

In the mean time I'll squishy merge this.

Reply:
makes sense.  I'd rather see this in a better place than buried in the cmake folder (because as a general rule... who looks there?)